src/main/resources/migration/common/V10__create_com_details_table.sql

```diff
@@ -1,8 +1,8 @@
 CREATE TABLE community_offender_manager
 (
-    id SERIAL NOT NULL PRIMARY KEY,
-    staff_identifier INTEGER NOT NULL UNIQUE,
-    username VARCHAR(40) NOT NULL UNIQUE,
+    id SERIAL NOT NULL constraint community_offender_manager_key PRIMARY KEY,
```

Comment:
this is modifying an existing migration which is usually a big no no - the problem is that we can't modify the unique staff identifier index in a migration as there is no name to refer to in integration tests.

H2 in memory db generates a random name for each constraint if not explicitly named,  whereas postgres defaults to a name with a known format.

This change is explicitly naming the constraint to be the same as postgres has defaulted to in production - so is effectively a no-op that only enables us to write migrations that work in integration tests.

Usually flyway validates all migration scripts contents against what has been previously run to try and detect this kind of nefariousness before running any new migrations. 

To get round this I've added a prevalidate script that tweaks the checksum in the DB to the value for the newly modified script. As mentioned above - this is safe as we know that specifiying the name of the index to be what postgres has called it anyway has no effect.

Reply:
I saw modifying a migration and grimaced but use case is fair having explored alternatives. Would have been nice if it allowed you to add a constraint such as this as a separate migration such as:

`ALTER TABLE community_offender_manager ADD CONSTRAINT constraint community_offender_manager_key UNIQUE (id);`

Reply:
you can.. but only if you know the name of the constraint which you don't in this case!  🤣